src/client.rs

```diff
@@ -18,119 +18,64 @@
 //! and parsing responses
 //!
 
-use std::{error, io};
+use std::borrow::Cow;
 use std::collections::HashMap;
-use std::sync::{Arc, Mutex};
+use std::sync::atomic;
 
 use serde;
-use base64;
-use http;
+// use base64;
 use serde_json;
 
 use super::{Request, Response};
 use util::HashableValue;
 use error::Error;
 
-/// An interface for an HTTP roundtripper that handles HTTP requests.
-pub trait HttpRoundTripper {
-    /// The type of the http::Response body.
-    type ResponseBody: io::Read;
-    /// The type for errors generated by the roundtripper.
-    type Err: error::Error;
+/// An interface for a transport over which to use the JSONRPC protocol.
+pub trait Transport {
+    /// The Error type for this transport.
+    /// Errors will get converted into Box<std::error::Error> so the
+    /// type here is not use any further.
```

Comment:
This is a confusing comment and can probably be dropped.

Reply:
FWIW, I think I understood it on a first read.